File: app/src/main/java/com/rk/xededitor/MainActivity/fragment/NoSwipeViewPager.java
package com.rk.xededitor.MainActivity.fragment;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.viewpager.widget.ViewPager;

public class NoSwipeViewPager extends ViewPager {

    private float startX;
    private float startY;

    public NoSwipeViewPager(Context context) {
        super(context);
    }

    public NoSwipeViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        // Only allow vertical scrolling
        switch (event.getAction()) {
            case MotionEvent.ACTION_DOWN:
                startX = event.getX();
                startY = event.getY();
                break;
            case MotionEvent.ACTION_MOVE:
                float dx = Math.abs(event.getX() - startX);
                float dy = Math.abs(event.getY() - startY);
                if (dy > dx) {
                    // Vertical scroll
                    return super.onTouchEvent(event);
                }
                break;
        }
        return false;
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        // Intercept touch event only for vertical scrolling
        switch (event.getAction()) {
            case MotionEvent.ACTION_DOWN:
                startX = event.getX();
                startY = event.getY();
                break;
            case MotionEvent.ACTION_MOVE:
                float dx = Math.abs(event.getX() - startX);
                float dy = Math.abs(event.getY() - startY);
                if (dy > dx) {
                    // Vertical scroll
                    return super.onInterceptTouchEvent(event);
                }
                break;
        }
        return false;
    }

}

File: app/src/main/java/com/rk/xededitor/MainActivity/editor/NoSwipeViewPager.kt
package com.rk.xededitor.MainActivity.editor

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager
import kotlin.math.abs

class NoSwipeViewPager : ViewPager {
    private var startX = 0f
    private var startY = 0f

    constructor(context: Context?) : super(context!!)

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    )

    override fun onTouchEvent(event: MotionEvent): Boolean {
        // Only allow vertical scrolling
        when (event.action) {
            MotionEvent.ACTION_DOWN -> {
                startX = event.x
                startY = event.y
            }

            MotionEvent.ACTION_MOVE -> {
                val dx = abs((event.x - startX).toDouble()).toFloat()
                val dy = abs((event.y - startY).toDouble()).toFloat()
                if (dy > dx) {
                    // Vertical scroll
                    return super.onTouchEvent(event)
                }
            }
        }
        return false
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        // Intercept touch event only for vertical scrolling
        when (event.action) {
            MotionEvent.ACTION_DOWN -> {
                startX = event.x
                startY = event.y
            }

            MotionEvent.ACTION_MOVE -> {
                val dx = abs((event.x - startX).toDouble()).toFloat()
                val dy = abs((event.y - startY).toDouble()).toFloat()
                if (dy > dx) {
                    // Vertical scroll
                    return super.onInterceptTouchEvent(event)
                }
            }
        }
        return false
    }
}

